### ISSUE TITLE
Custom Action Handler not invoked for specific flows

### ISSUE THREAD

**Issue by FDavidse20** [Iterable/iterable-android-sdk#470]
We have set up push notifications with Custom Action (for the Open action). 
We've implemented the necessary methods and followed the steps from the documentation for iOS and Android. 

For iOS it's working flawlessly, (test) push notifications can be received and interacted with. 
For Android it is working sort of. 
Push notifications can be received and interacted with. 

That means that the action handler is invoked when pressing a push notification. 
And the performSomeAction() is called.

`val config = IterableConfig.Builder()
            .setAutoPushRegistration(false)
            .setPushIntegrationName(pushIntegrationName)
            .setCustomActionHandler { action, _ ->
                action.type?.let {
                   //do something, open a detail page in the app based on the data send in the action
                  performSomeAction()
                }
                true
            }
            .build()
        IterableApi.initialize(
            application,
            apiKey,
            config
        )`

However, there is a very specific flow in which that function is no longer called.

Steps to reproduce:
- Run app
- log in (push notifications are enabled, email is set, etc.)
- receive push notification (doesn't matter if user is interacting with the app, or if the app is in the background)
- open push notification
- app opens (if not already opened) and custom action handler is executed and performSomeAction() called
- hard close the app
- receive push notification
- open push notification
- app opens and custom action handler is executed and performSomeAction() called
- Everything still going well, as expected, but now:
- receive push notification
- open push notification
- app opens (if not already opened) and custom action handler is NOT executed and performSomeAction() NOT called

in other words, after hard closing the app, the flow will only work perfectly once more, after that something is broken. 

I've tried all sorts of things debugging, but frankly haven't got a clue. 

I hope you can help me, thank you.
Filip

**Comment by Ayyanchira** [comment]
Thanks for reporting this with detailed steps to repro @FDavidse20. I will check if I can reproduce this and will get back on this one...

**Comment by FDavidse20** [comment]
> Thanks for reporting this with detailed steps to repro @FDavidse20. I will check if I can reproduce this and will get back on this one...

Hi @Ayyanchira, any updates on this issue?

**Comment by dmitry-funda** [comment]
Hi @Ayyanchira , could you please address this issue?

**Comment by tnortman-jabra** [comment]
Just commenting here to say I have the same issue +1
